### ISSUE TITLE
resource-stats-recorder complains about missing v1alpha1 resources

### ISSUE THREAD

**Issue by tsawada** [GoogleCloudPlatform/k8s-config-connector#2402]
### Checklist

- [X] I did not find a related open issue.
- [X] I did not find a solution in the troubleshooting guide: (https://cloud.google.com/config-connector/docs/troubleshooting)
- [X] If this issue is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Bug Description

After updating Config Connector from v1.102.0 to v1.120.1, we get following ERROR logs for all kinds of v1alpha1 cnrm resources, a few hundreds / minute:

```
setup "msg"="error recording metrics for CRD %v: %v" "error"="error listing objects for compute.cnrm.cloud.google.com/v1alpha1, Kind=ComputeRegionPerInstanceConfig: error listing objects: no matches for kind \"ComputeRegionPerInstanceConfig\" in version \"compute.cnrm.cloud.google.com/v1alpha1\"" "compute.cnrm.cloud.google.com/v1alpha1, Kind=ComputeRegionPerInstanceConfig"="(MISSING)"
```

We are not sure if this actually indicates any breakage. If not, we prefer to have less ERROR logs so that we can notice real errors easier.

### Additional Diagnostic Information

None

### Kubernetes Cluster Version

v1.28.0

### Config Connector Version

1.120.1

### Config Connector Mode

namespaced mode (default)

### Steps to reproduce the issue

Update to v1.120.1

**Comment by cheftako** [comment]
In which log are you seeing this?

**Comment by tsawada** [comment]
Thanks. stderr of a pod `cnrm-resource-stats-recorder-***-**`, container `recorder`.

**Comment by justinsb** [comment]
Should be fixed by #2631, which should ship in 1.122

/close
